### PR TITLE
[Modules] Refactored Network Application Gateway to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Refactored Network Application Gateway to new dependency approach

| Pipeline |
| - |
| [Network: ApplicationGateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.applicationgateways.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/applicationGateways/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,385 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/applicationGateways/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/applicationGateways/.test/common/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    identity: {
+        type: 'UserAssigned'
+        userAssignedIdentities: {
+            '${managedIdentity.id}': {}
+        }
+    }
+    properties: {
+        azPowerShellVersion: '3.0'
+        retentionInterval: 'P1D'
+        arguments: ' -KeyVaultName "${keyVault.name}" -CertName "${CertName}"'
```
Comment (eriqua) [inline modules/Microsoft.Network/applicationGateways/.test/common/dependencies.bicep:99]: Ir should work without spaces
```suggestion
        arguments: '-KeyVaultName "${keyVault.name}" -CertName "${CertName}"'
```

File: modules/Microsoft.Network/applicationGateways/.test/common/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    location: location
+    kind: 'AzurePowerShell'
+    identity: {
+        type: 'UserAssigned'
+        userAssignedIdentities: {
+            '${managedIdentity.id}': {}
+        }
+    }
+    properties: {
+        azPowerShellVersion: '3.0'
```
Comment (eriqua) [inline modules/Microsoft.Network/applicationGateways/.test/common/dependencies.bicep:97]: Can we go with something more recent?
```suggestion
        azPowerShellVersion: '8.0'
```

File: modules/Microsoft.Network/applicationGateways/.test/.scripts/New-Certificate.ps1 [hunk 4]
```diff
@@ -0,0 +1,40 @@
+﻿param(
```
Comment (eriqua) [inline modules/Microsoft.Network/applicationGateways/.test/.scripts/New-Certificate.ps1:1]: Adding a point to issue #2038 to improve deployment scripts (e.g. add synopsis). Outside of this PR.

File: modules/Microsoft.Network/applicationGateways/.test/.scripts/New-Certificate.ps1 [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        SecretContentType = 'application/x-pkcs12'
+        SubjectName       = 'CN=fabrikam.com'
+        IssuerName        = 'Self'
+        ValidityInMonths  = 12
+        ReuseKeyOnRenewal = $true
+    }
+    $certPolicy = New-AzKeyVaultCertificatePolicy @policyInputObject
+
+    $null = Add-AzKeyVaultCertificate -VaultName $KeyVaultName -Name $CertName -CertificatePolicy $certPolicy
+    Write-Verbose ('Initated creation of certificate [{0}] in key vault [{1}]' -f $CertName, $KeyVaultName) -Verbose
```
Comment (eriqua) [inline modules/Microsoft.Network/applicationGateways/.test/.scripts/New-Certificate.ps1:19]:
```suggestion
    Write-Verbose ('Initiated creation of certificate [{0}] in key vault [{1}]' -f $CertName, $KeyVaultName) -Verbose
```

File: modules/Microsoft.Network/applicationGateways/.test/common/dependencies.bicep (hunk 2 above)
Comment (MrMCake) [reply modules/Microsoft.Network/applicationGateways/.test/common/dependencies.bicep:99]: I'll test it, but I don't know if will work without the space

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**